newtmgr/bll/bll_common.go

```diff
@@ -53,18 +53,25 @@ func exchangeMtu(cln ble.Client, preferredMtu uint16) (uint16, error) {
 			break
 		}
 
-		// If macOS returned a value other than 23, then MTU exchange has
+		// If macOS returned a value higher than 23, then MTU exchange has
 		// completed.
-		if mtu != bledefs.BLE_ATT_MTU_DFLT {
+		if mtu > bledefs.BLE_ATT_MTU_DFLT {
 			break
 		}
 
 		// Otherwise, give the OS some time to perform the exchange.
-		log.Debugf("macOS reports an MTU of 23.  " +
+		log.Debugf("macOS reports an MTU of <=23.  " +
 			"Assume exchange hasn't completed; wait and requery.")
 		time.Sleep(time.Second)
 	}
 
+	// On some versions of mac (e.g. monterey) workaround with
+	// looping three times seems to not work anymore. If after
+	// performing three loops MTU is still <=23 than hardcode it to 185.
+	if mtu <= bledefs.BLE_ATT_MTU_DFLT {
```

Comment:
it is perfectly fine to end with default MTU after MTU exchange,  we should consider overriding MTu only if it is invalid ( less than default)

Reply:
Fixed it